dask/array/tests/test_array_core.py

```diff
@@ -2192,6 +2192,31 @@ def test_long_slice():
     assert_eq(d[8000:8200], x[8000:8200])
 
 
+@pytest.mark.parametrize("ndim", [0, 1, 2])
+def test_item_singleton(ndim):
+    for s in range(ndim):
+        a = np.ones(s * (1,))
+        d = da.from_array(a, chunks=1)
+        assert_eq(a.item(), d.item())
```

Comment:
Would it make sense to expand `assert_eq` to also check that the type of the two sides is equal after computing?

Reply:
FWICT that appears to already happening. Originally had a type mismatch before adding `astype(object)` in above, which this test caught. So added that in.